sycl/source/detail/scheduler/commands.cpp

```diff
@@ -333,9 +333,11 @@ void Command::waitForEvents(QueueImplPtr Queue,
 }
 
 Command::Command(CommandType Type, QueueImplPtr Queue)
-    : MQueue(std::move(Queue)), MType(Type) {
+    : MQueue(std::move(Queue)), MEvent(std::make_shared<detail::event_impl>(MQueue)),
+      MPreparedDepsEvents(MEvent->getPreparedDepsEvents()),
```

Comment:
Suggest adding a comment saying that it's safe to bind  these references to event's members because the Command should not outlive the event it is connected with.

Reply:
Comment added